content/sensu-core/0.29/guides/troubleshooting-sensu.md

```diff
@@ -0,0 +1,322 @@
+│ ├── cert.pem
+│ ├── keycert.p12
+│ ├── key.pem
+│ └── req.pem
+├── sensu_ca
+│ ├── cacert.cer
+│ ├── cacert.pem
+│ ├── certs
+│ │ ├── 01.pem
+│ │ └── 02.pem
+│ ├── index.txt
+│ ├── index.txt.attr
+│ ├── index.txt.attr.old
+│ ├── index.txt.old
+│ ├── openssl.cnf
+│ ├── private
+│ │ └── cakey.pem
+│ ├── serial
+│ └── serial.old
+├── server
+│ ├── cert.pem
+│ ├── keycert.p12
+│ ├── key.pem
+│ └── req.pem
+└── ssl_certs.sh{{< /highlight >}}
+
+So from that directory, we'll ensure that our server cert and key (read "cert and key that are being used inside of the RabbitMQ configuration") match
+
+{{< highlight shell >}}
+$ openssl x509 -noout -modulus -in server/cert.pem | openssl md5
+(stdin)= 32df80471e2d4e7d0453f60cfb66b2b2
+$ openssl rsa -noout -modulus -in server/key.pem | openssl md5
+(stdin)= 32df80471e2d4e7d0453f60cfb66b2b2{{< /highlight >}}
+
+And then the same with our client cert and key (cert and key being used inside of the Sensu configuration):
+
+{{< highlight shell >}}
+$ openssl x509 -noout -modulus -in client/cert.pem | openssl md5
+(stdin)= c2a6a5a28a629653741e7674c3b95b19
+$ openssl rsa -noout -modulus -in client/key.pem | openssl md5
+(stdin)= c2a6a5a28a629653741e7674c3b95b19{{< /highlight >}}
+
+_WARNING: Should the values here NOT match, you'll need to regenerate your cert/key pairs._
+
+And just to test to ensure that we can indeed connect with our cert/key pairs, we can use openssl to connect to our RabbitMQ instance directly:
+
+{{< highlight shell >}}
+$ openssl s_client -connect localhost:5671 -key client/key.pem{{< /highlight >}}
+
+Which should give you output that will look similar to the following:
+
+{{< highlight shell >}}
+CONNECTED(00000003)
+depth=1 CN = SensuCA
+verify error:num=19:self signed certificate in certificate chain
+---
+Certificate chain
+0 s:/CN=sensu/O=server
+i:/CN=SensuCA
+1 s:/CN=SensuCA
+i:/CN=SensuCA
+{{< /highlight >}}
+
+Provided that the MD5 sums match and we're able to connect to RabbitMQ via openssl, we can effectively rule out any issues with the certificates.
+
+Let's move on to looking at the certificate (in this case, the server certificate specifically) and see what we find there. You can use the following command to examine the inner details of a certificate:
+
+{{< highlight shell >}}
+openssl x509 -in server/cert.pem -text -noout{{< /highlight >}}
+
+This will give you quite a bit, but the most important thing to note here is a specific extension:
+
+{{< highlight shell >}}        
+        X509v3 extensions:
+            X509v3 Basic Constraints:
+                CA:FALSE
+            X509v3 Key Usage:
+                Key Encipherment
+            X509v3 Extended Key Usage:
+                TLS Web Server Authentication{{< /highlight >}}
+
+In the output above, we're specifically interested in the `TLS Web Server Authentication` extension. In a non-working certificate, _you will not see this present._ Instead, you'll end up seeing a value that looks similar to a SMTP MIB.
```

Comment:
Should be `SNMP MIB`. Example too? Or maybe just a work in progress thing you'll get to later, 👍.